btx/processing/peak_finder.py

```diff
@@ -304,36 +308,39 @@ def find_peaks(self):
 
         for idx in np.arange(start_idx, end_idx):
 
-            # retrieve calibrated image
+            # retrieve event and find out if we skip it
             evt = self.psi.runner.event(self.psi.times[idx])
-            self.psi.get_timestamp(evt.get(EventId))
-            img = self.psi.det.calib(evt=evt)
-            if img is None:
-                empty_images += 1
-                continue
+            if not self.psi.skip_event(evt):
```

Comment:
Would it be more efficient to only call `self.psi.skip_event` if `self.psi.event_logic==True`?

Reply:
Sorry this has been addressed now, in `skip_event()` which is always `False` if `self.event_receiver` is None.